src/rust/zh.manhuagui/src/parser.rs

```diff
@@ -0,0 +1,329 @@
+use crate::{
+	decoder::{decompress_from_base64, Decoder},
+	helper::{self, encode_uri, i32_to_string},
+};
+
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::html::Node,
+	std::Vec,
+	std::{net::HttpMethod, net::Request, String},
+	Chapter, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
+	MangaViewer, Page,
+};
+use alloc::vec;
+
+const BASE_URL: &str = "https://www.manhuagui.com";
+// const mobileURL: &str = "https://m.manhuagui.com";
+
+const FILTER_REGION: [&str; 7] = [
+	"all", "japan", "hongkong", "other", "europe", "china", "korea",
+];
+const FILTER_GENRE: [&str; 39] = [
+	"all",
+	"rexue",
+	"maoxian",
+	"mohuan",
+	"shengui",
+	"gaoxiao",
+	"mengxi",
+	"aiqing",
+	"kehuan",
+	"mofa",
+	"gedou",
+	"wuxia",
+	"jizhan",
+	"zhanzheng",
+	"jingji",
+	"tiyu",
+	"xiaoyuan",
+	"shenghuo",
+	"lizhi",
+	"lishi",
+	"weiniang",
+	"zhainan",
+	"funv",
+	"danmei",
+	"baihe",
+	"hougong",
+	"zhiyu",
+	"meishi",
+	"tuili",
+	"xuanyi",
+	"kongbu",
+	"sige",
+	"zhichang",
+	"zhentan",
+	"shehui",
+	"yinyue",
+	"wudao",
+	"zazhi",
+	"heidao",
+];
+const FILTER_AUDIENCE: [&str; 6] = [
+	"all", "shaonv", "shaonian", "qingnian", "ertong", "tongyong",
+];
+const FILTER_PROGRESS: [&str; 3] = ["all", "lianzai", "wanjie"];
+
+pub fn parse_home_page(html: Node) -> Result<MangaPageResult> {
+	let mut mangas: Vec<Manga> = Vec::new();
+
+	let ul = "#contList > li";
+	// let elements = html.select(ul).array();
+
+	for element in html.select(ul).array() {
+		let elem = element.as_node().unwrap();
```

Comment:
continue if this fails instead of panic
```suggestion
		let elem = match element.as_node() {
			Ok(node) => node,
			Err(_) => continue,
		};
```